include/deal.II/base/tensor.h

```diff
@@ -1793,7 +1793,7 @@ Iterator
 Tensor<rank_, dim, Number>::unroll_recursion(const Iterator current,
                                              const Iterator end) const
 {
-  auto next = current;
+  auto *next = current;
```

Comment:
This only works if `Iterator` is a pointer.

Reply:
I think this is probably better written as
```suggestion
  Iterator next = current;
```

Reply:
Indeed, MSVC complains:
```
D:\a\dealii\dealii\include\deal.II/base/tensor.h(1796,14): error C3535: cannot deduce type for 'auto *' from 'const Iterator' [D:\a\dealii\dealii\build\source\base\object_base_debug.vcxproj]
          with
          [
              Iterator=std::_Array_iterator<float,1>
          ]
```